google-cloud-firestore/lib/google/cloud/firestore/client.rb

```diff
@@ -185,6 +186,47 @@ def col_group collection_id
         end
         alias collection_group col_group
 
+        ##
+        # Creates a filter object.
+        #
+        # @param [FieldPath, String, Symbol] field A field path to filter
+        #   results with.
+        #
+        #   If a {FieldPath} object is not provided then the field will be
+        #   treated as a dotted string, meaning the string represents individual
+        #   fields joined by ".". Fields containing `~`, `*`, `/`, `[`, `]`, and
+        #   `.` cannot be in a dotted string, and should provided using a
+        #   {FieldPath} object instead.
+        #
+        # @param [String, Symbol] operator The operation to compare the field
+        #   to. Acceptable values include:
+        #
+        #   * less than: `<`, `lt`
+        #   * less than or equal: `<=`, `lte`
+        #   * greater than: `>`, `gt`
+        #   * greater than or equal: `>=`, `gte`
+        #   * equal: `=`, `==`, `eq`, `eql`, `is`
+        #   * not equal: `!=`
+        #   * in: `in`
+        #   * not in: `not-in`, `not_in`
+        #   * array contains: `array-contains`, `array_contains`
+        #
+        # @param [Object] value A value the field is compared to.
```

Comment:
Based on an earlier discussion, it might be helpful to list the valid datatypes under `Object`. You can mention them as a bulleted list, like [this](https://github.com/googleapis/google-cloud-ruby/pull/21002/files#diff-aa14a2bcf1c593d9dfe69ab1ff512ba309ce5617a0d2ef56d071ea6d237a3878R990-R1001).